test/Faker/Extension/ContainerBuilderTest.php

```diff
@@ -14,6 +15,99 @@
  */
 final class ContainerBuilderTest extends TestCase
 {
+    /**
+     * @dataProvider provideInvalidValue
+     *
+     * @param array|bool|float|int|resource|null $value
+     */
+    public function testAddRejectsInvalidValue($value): void
+    {
+        $containerBuilder = new ContainerBuilder();
+
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage(sprintf(
+            'First argument to "%s::add()" must be a string, callable or object.',
+            ContainerBuilder::class
+        ));
+
+        $containerBuilder->add($value);
+    }
+
+    /**
+     * @return \Generator<string, array{0: array|bool|float|int|null|resource}>
+     */
+    public function provideInvalidValue(): \Generator
+    {
+        $values = [
+            'array' => [
+                'foo',
+                'bar',
+                'baz',
+            ],
+            'bool-false' => false,
+            'bool-true' => true,
+            'float' => 3.14,
+            'int' => 9001,
+            'null' => true,
+            'resource' => fopen(__FILE__, 'rb'),
+        ];
+
+        foreach ($values as $key => $value) {
+            yield $key => [
+                $value,
+            ];
+        }
+    }
+
+    public function testAddRejectsNameWhenValueIsCallableAndNameIsNull(): void
+    {
+        $value = [
+            new class() {
+                public static function create(): Extension
+                {
+                    return new class() implements Extension {
+                        public function foo(): string
+                        {
+                            return 'foo';
+                        }
+                    };
+                }
+            },
+            'create',
+        ];
+
+        $containerBuilder = new ContainerBuilder();
+
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage(sprintf(
+            'Second argument to "%s::add()" is required not passing a string or object as first argument',
+            ContainerBuilder::class
+        ));
+
+        $containerBuilder->add($value);
+    }
+
+    public function testAddAcceptsValueWhenItIsAnObjectAndNameIsNull(): void
+    {
+        $value = new class() implements Extension {
+            public function foo(): string
```

Comment:
Why is the function foo() needed here?

Reply:
It is not needed, it is only to demonstrate that we implement an extension here. An empty extension would be useless, so I added a method `foo()` here. For the test it would suffice to not have the method.